osquery/tables/networking/posix/curl.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <chrono>
+
+#include <boost/network/include/http/client.hpp>
+#include <boost/numeric/conversion/cast.hpp>
+
+#include <osquery/logger.h>
+#include <osquery/status.h>
+#include <osquery/tables.h>
+
+using namespace boost::network;
+using namespace boost::network::http;
+using namespace std::chrono;
+
+namespace osquery {
+namespace tables {
+
+const std::string OSQUERY_USER_AGENT = "osquery";
+
+Status processRequest(const std::string& request_str, QueryData& results) {
+  Row r;
+  r["url"] = request_str;
+  r["method"] = "GET";
+  r["ua"] = OSQUERY_USER_AGENT;
```

Comment:
It looks like this is hardcoded above, but the table schema implies you can set it? Would it be possible to pass the 'ua' constraing from the query (if it exists) through to processRequest, so that if the user wants to specify a custom UA they can? Otherwise the default of osquery is used.

Reply:
+1